### PR TITLE
travis: add a test regarding commit messages

### PR DESCRIPTION
**Pull request by roidelapluie**
Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

### CODE REVIEW

File: test/test-commit-message.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+travis_regex='^[a-zA-Z0-9]\+: '
```
Comment (purpleidea) [inline test/test-commit-message.sh]: Prefix should be only lower case, and allow ', '  (comma space)
eg:

resources: file: Blah blah

or

pgraph, lib: Do thing

File: test/test-commit-message.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+if [[ -n "$TRAVIS_COMMIT_MESSAGE" ]]
+then
+    echo Testing commit message.
+    if ! echo "$TRAVIS_COMMIT_MESSAGE" | head -n 1 | grep -q "$travis_regex"
+    then
+        echo "Commit message should follow the following regex: '$travis_regex'"
+        echo
+        echo "e.g:"
+        echo "prometheus: implement rest api"
+        echo "Svc: fix a race condition with reloads"
```
Comment (purpleidea) [inline test/test-commit-message.sh]: s/Svc/svc/

Comment (roidelapluie) [inline test/test-commit-message.sh]: everywhere you specify resources as caps, am I right?

**Comment by purpleidea** [comment]
Build will fail b/c of:

FAIL: The following bash files are not properly formatted: test/test-commit-message.sh

You should indent with tabs, not spaces.
And did you want bash -x or bash -e at the top??

**Comment by purpleidea** [comment]
err `set -e rather`. I forget if that's the same as bash -e

**Comment by roidelapluie** [comment]
seriously? tabs in bash? O_o

**Comment by roidelapluie** [comment]
@purpleidea fixed

**Comment by purpleidea** [comment]
On Thu, Feb 9, 2017 at 9:03 AM, Julien Pivotto <notifications@github.com> wrote:
>
> seriously? tabs in bash? O_o

It's what all the cool kids are doing. Hey-- at least I have a test
case for it :P

**Comment by roidelapluie** [comment]
really wondering if that kind of tests are useful and welcoming new contribs (same as this PR)

**Comment by purpleidea** [comment]
On Thu, Feb 9, 2017 at 9:10 AM, Julien Pivotto <notifications@github.com>
wrote:

> really wondering if that kind of tests are useful and welcoming new
> contribs (same as this PR)



Enforcing good practices that tell the user automatically what to fix are
IMO a good thing to have.

**Comment by purpleidea** [comment]
This failed due to:
TRAVIS_COMMIT_MESSAGE=Merge 
                    705462aa32499b516226e67860119845ec759e70 into 
                    7b7c765d78b082c2d77d7dcbc74b1d91369e8c17
Maybe we should loop with git log?

**Comment by purpleidea** [comment]
Is this from this?
```
fatal: ambiguous argument 'origin/tcm..c00038e782416846f68c7e89cd637621799072bb': unknown revision or path not in the working tree.

Use '--' to separate paths from revisions, like this:

'git <command> [<revision>...] -- [<file>...]'
```

**Pull request merged**